### ISSUE TITLE
The safe-area insets are not enough to account for the navigation menu/status bar on Android

### ISSUE THREAD

**Issue by bulletbrand** [capacitor-community/safe-area#30]
**Describe the bug**
The capacitor-community/safe-area plugin works correctly on iOS but behaves strangely on Android. It enables edge-to-edge mode, but the insets at the top and bottom are incorrect, which results in a misaligned UI. The safe-area insets do not properly account for the navigation menu and status bar on Android.

On iOS, everything works as expected, but on Android, the insets do not consider the navigation bar, leading to incorrect spacing at the top and bottom of the screen, which causes the content to be displayed incorrectly.

**ANDROID:**
<img width="424" alt="" src="https://github.com/user-attachments/assets/b5e307c6-b714-4788-ada0-e45f6ba7d673">

**IOS**
<img width="514" alt="" src="https://github.com/user-attachments/assets/eda478cc-0a9f-4ebe-8d1b-1d20b04277ab">

**Version:**
API 30 - 35.

My config setup:
```
	"SafeArea": {
			"enabled": true,
			"customColorsForSystemBars": true,
			"statusBarColor": "#000000",
			"statusBarContent": "light",
			"navigationBarColor": "#000000",
			"navigationBarContent": "light"
		}
```

and css import in root file of course

A also tried with Plugin import option but the result is the same.

Should this work on iOS and correctly account for the top and bottom insets, or is this plugin intended to be used in combination with capacitor-safe-area?

**Comment by bulletbrand** [comment]
At the same time, I find it strange that when using Chrome Inspect, the inset looks more than sufficient. I would assume this is a glitch in the emulator's behavior, but everything works correctly without this plugin activated. It might be an effect of the edge-to-edge mode. Have you encountered anything similar?

<img width="384" alt="" src="https://github.com/user-attachments/assets/c86e2951-fbd1-4f78-9c2f-500db4ee8cdd">
tered anything similar?

This might be an issue specific to my environment, but I currently have no ideas why it's happening

**Comment by bulletbrand** [comment]
@tafelnl yes, sure
Something is clearly causing some kind of failure in the plugin, and from the screens you can see that it works, but at the same time gives incorrect indents. Moreover, currently I only see the issue with the **bottom** safe-area space.

I use **Capacitor 6** and **openjdk 17.0.4.1 2022-08-12**.

I’m trying to run with **Android Studio Koala 2024.1.1 Patch 2** on emulators. API version 30 - 35, any emulator.

My android setup:

**android/build.gradle**
```
buildscript {
    ext.kotlin_version = project.hasProperty("kotlin_version") ? rootProject.ext.kotlin_version : '1.9.10'

    repositories {
        google()
        mavenCentral()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:8.2.1'
        classpath 'com.google.gms:google-services:4.4.0'
    }
}

apply from: "variables.gradle"

allprojects {
    repositories {
        google()
        mavenCentral()
    }
}

task clean(type: Delete) {
    delete rootProject.buildDir
}

```

**android/app/build.gradlew**
```
apply plugin: 'com.android.application'
android {
    namespace "app"
    compileSdk rootProject.ext.compileSdkVersion
    defaultConfig {
        ...
    }
    signingConfigs {
        staging {
            ...
        }
        release {
            ...
        }
    }
    buildTypes {
           ...
        }
        staging {
           ...
        }
        release {
           ...
        }
    }
}

repositories {
    flatDir{
        dirs '../capacitor-cordova-android-plugins/src/main/libs', 'libs'
    }
}

dependencies {
    implementation "androidx.core:core-splashscreen:$coreSplashScreenVersion"
    implementation "androidx.coordinatorlayout:coordinatorlayout:$androidxCoordinatorLayoutVersion"
    implementation fileTree(include: ['*.jar'], dir: 'libs')
    implementation "androidx.appcompat:appcompat:$androidxAppCompatVersion"
    implementation project(':capacitor-android')
    testImplementation "junit:junit:$junitVersion"
    androidTestImplementation "androidx.test.ext:junit:$androidxJunitVersion"
    androidTestImplementation "androidx.test.espresso:espresso-core:$androidxEspressoCoreVersion"
    implementation project(':capacitor-cordova-android-plugins')
    implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
}

apply from: 'capacitor.build.gradle'

try {
    def servicesJSON = file('google-services.json')
    if (servicesJSON.text) {
        apply plugin: 'com.google.gms.google-services'
    }
} catch(Exception e) {
    logger.warn("google-services.json not found, google-services plugin not applied. Push Notifications won't work")
}

```

**capacitor.setting**

```
include ':capacitor-android'
project(':capacitor-android').projectDir = new File('../node_modules/@capacitor/android/capacitor')

include ':atroo-capacitor-secure-storage-plugin'
project(':atroo-capacitor-secure-storage-plugin').projectDir = new File('../node_modules/@atroo/capacitor-secure-storage-plugin/android')

include ':byteowls-capacitor-oauth2'
project(':byteowls-capacitor-oauth2').projectDir = new File('../node_modules/@byteowls/capacitor-oauth2/android')

include ':capacitor-community-barcode-scanner'
project(':capacitor-community-barcode-scanner').projectDir = new File('../node_modules/@capacitor-community/barcode-scanner/android')

include ':capacitor-community-file-opener'
project(':capacitor-community-file-opener').projectDir = new File('../node_modules/@capacitor-community/file-opener/android')

include ':capacitor-community-safe-area'
project(':capacitor-community-safe-area').projectDir = new File('../node_modules/@capacitor-community/safe-area/android')

include ':capacitor-community-sqlite'
project(':capacitor-community-sqlite').projectDir = new File('../node_modules/@capacitor-community/sqlite/android')

include ':capacitor-firebase-messaging'
project(':capacitor-firebase-messaging').projectDir = new File('../node_modules/@capacitor-firebase/messaging/android')

include ':capacitor-app'
project(':capacitor-app').projectDir = new File('../node_modules/@capacitor/app/android')

include ':capacitor-app-launcher'
project(':capacitor-app-launcher').projectDir = new File('../node_modules/@capacitor/app-launcher/android')

include ':capacitor-browser'
project(':capacitor-browser').projectDir = new File('../node_modules/@capacitor/browser/android')

include ':capacitor-camera'
project(':capacitor-camera').projectDir = new File('../node_modules/@capacitor/camera/android')

include ':capacitor-clipboard'
project(':capacitor-clipboard').projectDir = new File('../node_modules/@capacitor/clipboard/android')

include ':capacitor-device'
project(':capacitor-device').projectDir = new File('../node_modules/@capacitor/device/android')

include ':capacitor-filesystem'
project(':capacitor-filesystem').projectDir = new File('../node_modules/@capacitor/filesystem/android')

include ':capacitor-geolocation'
project(':capacitor-geolocation').projectDir = new File('../node_modules/@capacitor/geolocation/android')

include ':capacitor-haptics'
project(':capacitor-haptics').projectDir = new File('../node_modules/@capacitor/haptics/android')

include ':capacitor-keyboard'
project(':capacitor-keyboard').projectDir = new File('../node_modules/@capacitor/keyboard/android')

include ':capacitor-live-updates'
project(':capacitor-live-updates').projectDir = new File('../node_modules/@capacitor/live-updates/android')

include ':capacitor-local-notifications'
project(':capacitor-local-notifications').projectDir = new File('../node_modules/@capacitor/local-notifications/android')

include ':capacitor-network'
project(':capacitor-network').projectDir = new File('../node_modules/@capacitor/network/android')

include ':capacitor-preferences'
project(':capacitor-preferences').projectDir = new File('../node_modules/@capacitor/preferences/android')

include ':capacitor-screen-orientation'
project(':capacitor-screen-orientation').projectDir = new File('../node_modules/@capacitor/screen-orientation/android')

include ':capacitor-share'
project(':capacitor-share').projectDir = new File('../node_modules/@capacitor/share/android')

include ':capacitor-splash-screen'
project(':capacitor-splash-screen').projectDir = new File('../node_modules/@capacitor/splash-screen/android')

include ':capacitor-status-bar'
project(':capacitor-status-bar').projectDir = new File('../node_modules/@capacitor/status-bar/android')

```

package.json ( app related )

 ```
"dependencies": {
    "@byteowls/capacitor-oauth2": "^4.0.0",
    "@capacitor-community/barcode-scanner": "^4.0.1",
    "@capacitor-community/file-opener": "^1.0.5",
    **"@capacitor-community/safe-area": "^6.0.0-alpha.6",**
    "@capacitor-community/sqlite": "^5.2.3",
    "@capacitor-firebase/messaging": "^5.1.0",
    "@capacitor/android": "^6.0.0",
    "@capacitor/app": "^6.0.0",
    "@capacitor/app-launcher": "^6.0.0",
    "@capacitor/browser": "^6.0.0",
    "@capacitor/camera": "^6.0.0",
    "@capacitor/clipboard": "^6.0.0",
    "@capacitor/core": "^6.0.0",
    "@capacitor/device": "^6.0.0",
    "@capacitor/filesystem": "^6.0.0",
    "@capacitor/geolocation": "^6.0.0",
    "@capacitor/haptics": "^6.0.0",
    "@capacitor/ios": "^6.0.0",
    "@capacitor/keyboard": "^6.0.0",
    "@capacitor/live-updates": "^0.2.1",
    "@capacitor/local-notifications": "^6.0.0",
    "@capacitor/network": "^6.0.0",
    "@capacitor/preferences": "^6.0.0",
    "@capacitor/screen-orientation": "^6.0.0",
    "@capacitor/share": "^6.0.0",
    "@capacitor/splash-screen": "^6.0.0",
    "@capacitor/status-bar": "^6.0.0",
    "env-cmd": "^10.1.0"
  }
```

I believe this information should be sufficient. I’ll try adding the package to a new app, as I’m running out of ideas on why the emulator might be acting this way.
I have suspicions that this problem is reproduced exclusively on emulators, but I don't have a real android device to check it yet.

**Comment by bulletbrand** [comment]
@tafelnl sure, I will do

**Comment by bulletbrand** [comment]
@tafelnl 

**Demo app with issue** :( just empty react/capacitor app ):
https://github.com/bulletbrand/safe-area-demo-repo

1)You might try running a **Pixel 8 emulator with API 30**. After executing the two commands from the README, you'll see a demo application. In this app, despite the safe-area-bottom being applied and showing an indentation of 26px, the bottom menu still gets hidden under the system's bottom navigation bar.

I suspect that this behavior might be related to the edge-to-edge mode, particularly on emulators. However, I haven't been able to confirm this. In any case, there does seem to be an issue present. ( I suspect that the height of the system menu itself is not being taken into account, which is causing the issue. )

It's worth noting that everything works fine on iOS.

Expected behavior:
<img width="405" alt="" src="https://github.com/user-attachments/assets/0221f787-0d36-4f08-b7fc-36468d7ae50b">

Current behavior:
<img width="407" alt="" src="https://github.com/user-attachments/assets/b908c865-a0e0-49f0-9ae1-2cfdf9954ea5">

2)
I also observed that the plugin functions properly only when **JDK 17** is installed. When I tried using **JDK 18**, I encountered conflicts that I couldn't resolve using standard methods. As a result, I had to downgrade Java to version 17 (this was tested on two different machines: Windows and macOS).
